midend/interpreter.cpp

```diff
@@ -866,9 +866,11 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
     if (l->is<SymbolicInteger>()) {
         auto li = l->to<SymbolicInteger>();
         clone->expr = li->constant;
+        auto type = typeMap->getType(getOriginal(), true);
```

Comment:
I think that we need to do the same thing for other cases.

Reply:
The reason we don't need to set the type in the typemap in the other cases is because the constant folding in general does not need to know types - but it does need to know if for casts.